.github/workflows/build.yaml

```
@@ -0,0 +1,201 @@
+  # │   BUILD JOB                                                                                                        │
+  # │   ---                                                                                                              │
+  # │                                                                                                                    │
+  # │   The 'build' job compiles and packages the KSSO py application for different OS and Python versions.              │
+  # │   It uses a matrix strategy to run builds.                                                                         │
+  # │   The reason we need a matrix build, because we are using pyinstaller to produce executable binary                 │
+  # │   and in order to do so, it should be executed in the target OS.                                                   │
+  # │                                                                                                                    │
+  # │   Outputs:                                                                                                         │
+  # │   - `version`: Extracted from the project version and passed to the `release` job.                                 │
+  # │                                                                                                                    │
+  # └────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+  build:
+    name: Build
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macos-latest
+          - ubuntu-latest
+          - windows-latest
+          - macos-13
+        python-version:
+          - 3.12
+    timeout-minutes: 30
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Set up python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Cache poetry dependencies
+        id: cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/pypoetry
+            ~/Library/Caches/pypoetry
+            C:\Users\runneradmin\AppData\Local\pypoetry\Cache
+          key: ${{ runner.os }}-${{ runner.arch }}-poetry-${{ hashFiles('**/poetry.lock')}}
+          restore-keys: |
+            ${{ runner.os }}-${{ runner.arch }}-poetry-
+      - name: Install poetry
+        run: |
+          python -m pip install poetry
+      - name: Install dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: poetry install --no-interaction --no-ansi
+      - name: Determine version
+        id: version
+        run: |
+          version=$(poetry version --short)
+          echo "version=$version" >> $GITHUB_ENV
+          echo "::set-output name=version::$version"
```

Can we use newer GITHUB_ENV variable instead?
    
```suggestion
          echo "version=$version" >> $GITHUB_ENV
```

---

we need the older implementation due to the reference of **version** in the **release** job
```
 - name: Create release
        uses: actions/create-release@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          tag_name: v${{ needs.build.outputs.version }}
          release_name: ksso v${{ needs.build.outputs.version }}
          draft: false
          prerelease: false
```
which is not working with pure GITHUB_ENV

---

GITHUB_OUTPUT is also available, but you may find the same issue:

`echo "{name}={value}" >> "$GITHUB_OUTPUT"`

---

Ok let me try

---

it worked! Thank you Luciano!